File: android/app/src/debug/java/com/laneshadow/sandbox/mockproviders/PlanningMockProvider.kt
package com.laneshadow.sandbox.mockproviders

import com.laneshadow.ui.atoms.PhaseDotState
import com.laneshadow.ui.planning.PlanningPhaseStep

/**
 * Mock provider for Planning screen data.
 *
 * Provides planning phases, navigator message, and thinking state
 * for the Planning/route-building screen.
 *
 * PLAN-S08-AND-T02: Updated to provide capsuleHeadline, phaseSteps, and headerLabel
 * to support the PlanningScreen composition with LSContextCapsule and LSPhaseIndicator.
 */
object PlanningMockProvider : MockProvider<PlanningScreenState> {

    override val variants: List<String> = listOf(
        "default", "empty", "overflow", "long-copy",
        "scouting", "drawing", "weather", "scoring",
        "v-slow", "v-cancel-confirm", "v-single-candidate"
    )

    override fun value(variant: String): PlanningScreenState {
        return when (variant) {
            "empty" -> emptyState()
            "overflow" -> overflowState()
            "long-copy" -> longCopyState()
            "scouting" -> scoutingState()
            "drawing" -> drawingState()
            "weather" -> weatherState()
            "scoring" -> scoringState()
            "v-slow" -> slowState()
            "v-cancel-confirm" -> cancelConfirmState()
            "v-single-candidate" -> singleCandidateState()
            else -> defaultState()
        }
    }

    private fun defaultState(): PlanningScreenState {
        return PlanningScreenState(
            phases = listOf(
                PlanningPhase(id = "parsing", label = "Parsing your request", status = "done"),
                PlanningPhase(id = "searching", label = "Searching for routes", status = "done"),
                PlanningPhase(id = "drafting", label = "Drafting options", status = "active"),
                PlanningPhase(id = "enriching", label = "Enriching details", status = "pending"),
                PlanningPhase(id = "finalizing", label = "Finalizing plan", status = "pending")
            ),
            message = NavigatorMessage(
                id = "msg-planning-001",
                sessionId = "session-001",
                body = "Scenic 2-hour ride, avoid highways",
                timestamp = "2026-04-25T10:30:00Z",
                kind = "prompt",
                attachments = null,
                detail = null,
                pinned = false
            ),
            isThinking = true,
            capsuleHeadline = "Refining…",
            phaseSteps = listOf(
                PlanningPhaseStep(id = "parsing", label = "Parsing your request", state = PhaseDotState.Done),
                PlanningPhaseStep(id = "searching", label = "Searching for routes", state = PhaseDotState.Done),
                PlanningPhaseStep(id = "drafting", label = "Drafting options", state = PhaseDotState.Active),
                PlanningPhaseStep(id = "enriching", label = "Enriching details", state = PhaseDotState.Pending),
                PlanningPhaseStep(id = "finalizing", label = "Finalizing plan", state = PhaseDotState.Pending)
            ),
            headerLabel = "Let me think on that…"
        )
    }

    private fun emptyState(): PlanningScreenState {
        return PlanningScreenState(
            phases = emptyList(),
            message = NavigatorMessage(
                id = "msg-planning-empty",
                sessionId = "session-empty",
                body = "",
                timestamp = "2026-04-25T10:30:00Z",
                kind = "prompt",
                attachments = null,
                detail = null,
                pinned = false
            ),
            isThinking = false,
            capsuleHeadline = "Sketching…",
            phaseSteps = emptyList(),
            headerLabel = "Let me think on that…"
        )
    }

    private fun overflowState(): PlanningScreenState {
        return PlanningScreenState(
            phases = listOf(
                PlanningPhase(id = "parsing", label = "Parsing your request", status = "done"),
                PlanningPhase(id = "searching", label = "Searching for routes", status = "done"),
                PlanningPhase(id = "drafting", label = "Drafting options", status = "done"),
                PlanningPhase(id = "enriching", label = "Enriching details", status = "active"),
                PlanningPhase(id = "finalizing", label = "Finalizing plan", status = "pending")
            ),
            message = NavigatorMessage(
                id = "msg-planning-overflow",
                sessionId = "session-overflow",
                body = "Scenic 2-hour ride, avoid highways, include stops at viewpoints and coffee shops",
                timestamp = "2026-04-25T10:30:00Z",
                kind = "prompt",
                attachments = null,
                detail = null,
                pinned = false
            ),
            isThinking = true,
            capsuleHeadline = "Scoring…",
            phaseSteps = listOf(
                PlanningPhaseStep(id = "parsing", label = "Parsing your request", state = PhaseDotState.Done),
                PlanningPhaseStep(id = "searching", label = "Searching for routes", state = PhaseDotState.Done),
                PlanningPhaseStep(id = "drafting", label = "Drafting options", state = PhaseDotState.Done),
                PlanningPhaseStep(id = "enriching", label = "Enriching details", state = PhaseDotState.Active),
                PlanningPhaseStep(id = "finalizing", label = "Finalizing plan", state = PhaseDotState.Pending)
            ),
            headerLabel = "Let me think on that…"
        )
    }

    private fun longCopyState(): PlanningScreenState {
        return PlanningScreenState(
            phases = listOf(
                PlanningPhase(
                    id = "parsing",
                    label = "Parsing your ride request and understanding your preferences",
                    status = "done"
                ),
                PlanningPhase(
                    id = "searching",
                    label = "Searching for potential road routes through the Santa Cruz Mountains",
                    status = "done"
                ),
                PlanningPhase(
                    id = "drafting",
                    label = "Drafting route options and checking that all segments connect properly",
                    status = "active"
                ),
                PlanningPhase(
                    id = "enriching",
                    label = "Enriching with weather conditions and scenic details",
                    status = "pending"
                ),
                PlanningPhase(
                    id = "finalizing",
                    label = "Finalizing plan based on scenery, difficulty, and weather",
                    status = "pending"
                )
            ),
            message = NavigatorMessage(
                id = "msg-planning-long",
                sessionId = "session-long",
                body = "I am looking for a truly scenic and memorable two-hour motorcycle ride that takes me down the coast to Santa Cruz while completely avoiding all highways and major roads. I want to see the ocean, ride through redwood forests, and maybe stop for coffee along the way.",
                timestamp = "2026-04-25T10:30:00Z",
                kind = "prompt",
                attachments = null,
                detail = null,
                pinned = false
            ),
            isThinking = true,
            capsuleHeadline = "Refining…",
            phaseSteps = listOf(
                PlanningPhaseStep(id = "parsing", label = "Parsing your ride request and understanding your preferences", state = PhaseDotState.Done),
                PlanningPhaseStep(id = "searching", label = "Searching for potential road routes through the Santa Cruz Mountains", state = PhaseDotState.Done),
                PlanningPhaseStep(id = "drafting", label = "Drafting route options and checking that all segments connect properly", state = PhaseDotState.Active),
                PlanningPhaseStep(id = "enriching", label = "Enriching with weather conditions and scenic details", state = PhaseDotState.Pending),
                PlanningPhaseStep(id = "finalizing", label = "Finalizing plan based on scenery, difficulty, and weather", state = PhaseDotState.Pending)
            ),
            headerLabel = "Let me think on that…"
        )
    }

    /**
     * S01: Scouting
     * Phase 1 (Scouting) active with pulsing ring; phases 2-5 pending
     */
    private fun scoutingState(): PlanningScreenState {
        return PlanningScreenState(
            phases = listOf(
                PlanningPhase(id = "parsing", label = "Scouting trails near you…", status = "active"),
                PlanningPhase(id = "searching", label = "Searching for routes", status = "pending"),
                PlanningPhase(id = "drafting", label = "Drafting options", status = "pending"),
                PlanningPhase(id = "enriching", label = "Enriching details", status = "pending"),
                PlanningPhase(id = "finalizing", label = "Finalizing plan", status = "pending")
            ),
            message = NavigatorMessage(
                id = "msg-planning-scouting",
                sessionId = "session-scouting",
                body = "Scenic 2-hour ride, avoid highways",
                timestamp = "2026-04-25T10:30:00Z",
                kind = "prompt",
                attachments = null,
                detail = null,
                pinned = false
            ),
            isThinking = true,
            capsuleHeadline = "Scouting trails near you…",
            phaseSteps = listOf(
                PlanningPhaseStep(id = "parsing", label = "Scouting trails near you…", state = PhaseDotState.Active),
                PlanningPhaseStep(id = "searching", label = "Searching for routes", state = PhaseDotState.Pending),
                PlanningPhaseStep(id = "drafting", label = "Drafting options", state = PhaseDotState.Pending),
                PlanningPhaseStep(id = "enriching", label = "Enriching details", state = PhaseDotState.Pending),
                PlanningPhaseStep(id = "finalizing", label = "Finalizing plan", state = PhaseDotState.Pending)
            ),
            headerLabel = "Let me think on that…"
        )
    }

    /**
     * S02: Drawing
     * Phase 1 done (check); phase 2 (Drawing) active; phases 3-5 pending
     */
    private fun drawingState(): PlanningScreenState {
        return PlanningScreenState(
            phases = listOf(
                PlanningPhase(id = "parsing", label = "Parsing your request", status = "done"),
                PlanningPhase(id = "searching", label = "Drawing the route…", status = "active"),
                PlanningPhase(id = "drafting", label = "Drafting options", status = "pending"),
                PlanningPhase(id = "enriching", label = "Enriching details", status = "pending"),
                PlanningPhase(id = "finalizing", label = "Finalizing plan", status = "pending")
            ),
            message = NavigatorMessage(
                id = "msg-planning-drawing",
                sessionId = "session-drawing",
                body = "Scenic 2-hour ride, avoid highways",
                timestamp = "2026-04-25T10:30:00Z",
                kind = "prompt",
                attachments = null,
                detail = null,
                pinned = false
            ),
            isThinking = true,
            capsuleHeadline = "Drawing the route…",
            phaseSteps = listOf(
                PlanningPhaseStep(id = "parsing", label = "Parsing your request", state = PhaseDotState.Done),
                PlanningPhaseStep(id = "searching", label = "Drawing the route…", state = PhaseDotState.Active),
                PlanningPhaseStep(id = "drafting", label = "Drafting options", state = PhaseDotState.Pending),
                PlanningPhaseStep(id = "enriching", label = "Enriching details", state = PhaseDotState.Pending),
                PlanningPhaseStep(id = "finalizing", label = "Finalizing plan", state = PhaseDotState.Pending)
            ),
            headerLabel = "Let me think on that…"
        )
    }

    /**
     * S03: Weather
     * Phases 1-2 done; phase 3 (Weather) active; phases 4-5 pending
     */
    private fun weatherState(): PlanningScreenState {
        return PlanningScreenState(
            phases = listOf(
                PlanningPhase(id = "parsing", label = "Parsing your request", status = "done"),
                PlanningPhase(id = "searching", label = "Searching for routes", status = "done"),
                PlanningPhase(id = "drafting", label = "Checking weather…", status = "active"),
                PlanningPhase(id = "enriching", label = "Enriching details", status = "pending"),
                PlanningPhase(id = "finalizing", label = "Finalizing plan", status = "pending")
            ),
            message = NavigatorMessage(
                id = "msg-planning-weather",
                sessionId = "session-weather",
                body = "Scenic 2-hour ride, avoid highways",
                timestamp = "2026-04-25T10:30:00Z",
                kind = "prompt",
                attachments = null,
                detail = null,
                pinned = false
            ),
            isThinking = true,
            capsuleHeadline = "Checking weather…",
            phaseSteps = listOf(
                PlanningPhaseStep(id = "parsing", label = "Parsing your request", state = PhaseDotState.Done),
                PlanningPhaseStep(id = "searching", label = "Searching for routes", state = PhaseDotState.Done),
                PlanningPhaseStep(id = "drafting", label = "Checking weather…", state = PhaseDotState.Active),
                PlanningPhaseStep(id = "enriching", label = "Enriching details", state = PhaseDotState.Pending),
                PlanningPhaseStep(id = "finalizing", label = "Finalizing plan", state = PhaseDotState.Pending)
            ),
            headerLabel = "Let me think on that…"
        )
    }

    /**
     * S04: Scoring
     * Phases 1-3 done; phase 4 (Scoring) active; phase 5 pending
     * Note: The "overflow" variant currently represents this state; reusing it here
     */
    private fun scoringState(): PlanningScreenState {
        return PlanningScreenState(
            phases = listOf(
                PlanningPhase(id = "parsing", label = "Parsing your request", status = "done"),
                PlanningPhase(id = "searching", label = "Searching for routes", status = "done"),
                PlanningPhase(id = "drafting", label = "Drafting options", status = "done"),
                PlanningPhase(id = "enriching", label = "Scoring candidates…", status = "active"),
                PlanningPhase(id = "finalizing", label = "Finalizing plan", status = "pending")
            ),
            message = NavigatorMessage(
                id = "msg-planning-scoring",
                sessionId = "session-scoring",
                body = "Scenic 2-hour ride, avoid highways, include stops at viewpoints and coffee shops",
                timestamp = "2026-04-25T10:30:00Z",
                kind = "prompt",
                attachments = null,
                detail = null,
                pinned = false
            ),
            isThinking = true,
            capsuleHeadline = "Scoring candidates…",
            phaseSteps = listOf(
                PlanningPhaseStep(id = "parsing", label = "Parsing your request", state = PhaseDotState.Done),
                PlanningPhaseStep(id = "searching", label = "Searching for routes", state = PhaseDotState.Done),
                PlanningPhaseStep(id = "drafting", label = "Drafting options", state = PhaseDotState.Done),
                PlanningPhaseStep(id = "enriching", label = "Scoring candidates…", state = PhaseDotState.Active),
                PlanningPhaseStep(id = "finalizing", label = "Finalizing plan", state = PhaseDotState.Pending)
            ),
            headerLabel = "Let me think on that…"
        )
    }

    /**
     * V01: slow variant
     * Italic apology message + dashed border
     */
    private fun slowState(): PlanningScreenState {
        return PlanningScreenState(
            phases = listOf(
                PlanningPhase(id = "parsing", label = "Parsing your request", status = "done"),
                PlanningPhase(id = "searching", label = "Searching for routes", status = "active"),
                PlanningPhase(id = "drafting", label = "Drafting options", status = "pending"),
                PlanningPhase(id = "enriching", label = "Enriching details", status = "pending"),
                PlanningPhase(id = "finalizing", label = "Finalizing plan", status = "pending")
            ),
            message = NavigatorMessage(
                id = "msg-planning-slow",
                sessionId = "session-slow",
                body = "Scenic 2-hour ride, avoid highways",
                timestamp = "2026-04-25T10:30:00Z",
                kind = "prompt",
                attachments = null,
                detail = null,
                pinned = false
            ),
            isThinking = true,
            capsuleHeadline = "Asking…",
            phaseSteps = listOf(
                PlanningPhaseStep(id = "parsing", label = "Parsing your request", state = PhaseDotState.Done),
                PlanningPhaseStep(id = "searching", label = "Searching for routes", state = PhaseDotState.Active),
                PlanningPhaseStep(id = "drafting", label = "Drafting options", state = PhaseDotState.Pending),
                PlanningPhaseStep(id = "enriching", label = "Enriching details", state = PhaseDotState.Pending),
                PlanningPhaseStep(id = "finalizing", label = "Finalizing plan", state = PhaseDotState.Pending)
            ),
            headerLabel = "Let me think on that…",
            slowApology = "Taking a moment to think"
        )
    }

    /**
     * V02: cancel-confirm variant
     * Show cancel confirmation modal
     */
    private fun cancelConfirmState(): PlanningScreenState {
        return PlanningScreenState(
            phases = listOf(
                PlanningPhase(id = "parsing", label = "Parsing your request", status = "done"),
                PlanningPhase(id = "searching", label = "Searching for routes", status = "done"),
                PlanningPhase(id = "drafting", label = "Drafting options", status = "active"),
                PlanningPhase(id = "enriching", label = "Enriching details", status = "pending"),
                PlanningPhase(id = "finalizing", label = "Finalizing plan", status = "pending")
            ),
            message = NavigatorMessage(
                id = "msg-planning-cancel",
                sessionId = "session-cancel",
                body = "Scenic 2-hour ride, avoid highways",
                timestamp = "2026-04-25T10:30:00Z",
                kind = "prompt",
                attachments = null,
                detail = null,
                pinned = false
            ),
            isThinking = true,
            capsuleHeadline = "Refining…",
            phaseSteps = listOf(
                PlanningPhaseStep(id = "parsing", label = "Parsing your request", state = PhaseDotState.Done),
                PlanningPhaseStep(id = "searching", label = "Searching for routes", state = PhaseDotState.Done),
                PlanningPhaseStep(id = "drafting", label = "Drafting options", state = PhaseDotState.Active),
                PlanningPhaseStep(id = "enriching", label = "Enriching details", state = PhaseDotState.Pending),
                PlanningPhaseStep(id = "finalizing", label = "Finalizing plan", state = PhaseDotState.Pending)
            ),
            headerLabel = "Let me think on that…",
            showCancelConfirm = true
        )
    }

    /**
     * V03: single-candidate variant
     * Warning border + phase headers
     */
    private fun singleCandidateState(): PlanningScreenState {
        return PlanningScreenState(
            phases = listOf(
                PlanningPhase(id = "parsing", label = "Parsing your request", status = "done"),
                PlanningPhase(id = "searching", label = "Searching for routes", status = "done"),
                PlanningPhase(id = "drafting", label = "Drafting options", status = "active")
            ),
            message = NavigatorMessage(
                id = "msg-planning-single",
                sessionId = "session-single",
                body = "Scenic 2-hour ride, avoid highways",
                timestamp = "2026-04-25T10:30:00Z",
                kind = "prompt",
                attachments = null,
                detail = null,
                pinned = false
            ),
            isThinking = true,
            capsuleHeadline = "Refining…",
            phaseSteps = listOf(
                PlanningPhaseStep(id = "parsing", label = "Parsing your request", state = PhaseDotState.Done),
                PlanningPhaseStep(id = "searching", label = "Searching for routes", state = PhaseDotState.Done),
                PlanningPhaseStep(id = "drafting", label = "Drafting options", state = PhaseDotState.Active)
            ),
            headerLabel = "Let me think on that…",
            warningBorder = true,
            phaseHeaders = mapOf(
                "parsing" to "Checking your preferences",
                "searching" to "Finding the best roads",
                "drafting" to "Refining the route"
            )
        )
    }
}

File: ios/LaneShadow/Sandbox/MockProviders/PlanningMockProvider.swift
import Foundation

/// Mock provider for Planning screen data.
///
/// Provides planning phases, navigator message, and thinking state
/// for the Planning/route-building screen.
public enum PlanningMockProvider: MockProvider {
    public static let variants = [
        "default",
        "empty",
        "overflow",
        "long-copy",
        "v-slow",
        "v-cancel-confirm",
        "v-single-candidate",
    ]

    public static func value(variant: String = "default") -> PlanningScreenState {
        switch variant {
        case "empty":
            emptyState()
        case "overflow":
            overflowState()
        case "long-copy":
            longCopyState()
        case "v-slow":
            slowPlanningState()
        case "v-cancel-confirm":
            cancelConfirmState()
        case "v-single-candidate":
            singleCandidateState()
        default:
            defaultState()
        }
    }

    private static func defaultState() -> PlanningScreenState {
        PlanningScreenState(
            phases: [
                PlanningPhaseData(id: "parsing", label: "Parsing your request", status: "done"),
                PlanningPhaseData(id: "searching", label: "Searching routes", status: "active"),
                PlanningPhaseData(id: "drafting", label: "Drafting options", status: "pending"),
                PlanningPhaseData(id: "enriching", label: "Enriching with details", status: "pending"),
                PlanningPhaseData(id: "finalizing", label: "Finalizing your routes", status: "pending"),
            ],
            message: NavigatorMessage(
                id: "msg-001",
                sessionId: "session-001",
                body: "Looking for the best twisty roads between Santa Cruz and Big Sur...",
                timestamp: "2025-04-25T10:30:00Z",
                kind: "response",
                attachments: nil,
                detail: nil,
                pinned: false
            ),
            isThinking: true,
            showSlowApology: false,
            showCancelConfirm: false,
            showWarningChrome: false
        )
    }

    private static func emptyState() -> PlanningScreenState {
        PlanningScreenState(
            phases: [],
            message: NavigatorMessage(
                id: "msg-001",
                sessionId: "session-001",
                body: "Starting to plan your ride...",
                timestamp: "2025-04-25T10:30:00Z",
                kind: "response",
                attachments: nil,
                detail: nil,
                pinned: false
            ),
            isThinking: false,
            showSlowApology: false,
            showCancelConfirm: false,
            showWarningChrome: false
        )
    }

    private static func overflowState() -> PlanningScreenState {
        PlanningScreenState(
            phases: [
                PlanningPhaseData(id: "parsing", label: "Parsing your ride request", status: "done"),
                PlanningPhaseData(id: "searching", label: "Searching terrain", status: "done"),
                PlanningPhaseData(id: "drafting", label: "Drafting primary routes", status: "done"),
                PlanningPhaseData(id: "enriching", label: "Enriching with alternatives", status: "active"),
                PlanningPhaseData(id: "finalizing", label: "Finalizing recommendations", status: "pending"),
            ],
            message: NavigatorMessage(
                id: "msg-001",
                sessionId: "session-001",
                body: "I'm analyzing every possible route combination between Santa Cruz and Big Sur, taking into account road conditions, weather patterns, traffic forecasts, scenic value, and your personal preferences for twisty roads and coastal views. This is taking longer than usual because there are so many excellent options in this area.",
                timestamp: "2025-04-25T10:30:00Z",
                kind: "response",
                attachments: nil,
                detail: "I've found 47 potential routes so far and am evaluating each one.",
                pinned: false
            ),
            isThinking: true,
            showSlowApology: false,
            showCancelConfirm: false,
            showWarningChrome: false
        )
    }

    private static func longCopyState() -> PlanningScreenState {
        PlanningScreenState(
            phases: [
                PlanningPhaseData(
                    id: "parsing",
                    label: "Parsing and understanding your comprehensive ride request with all specified parameters and preferences",
                    status: "done"
                ),
                PlanningPhaseData(
                    id: "searching",
                    label: "Searching preliminary route options based on topographical analysis and road network mapping",
                    status: "active"
                ),
                PlanningPhaseData(
                    id: "drafting",
                    label: "Drafting road surface conditions, closure status, and accessibility for all route segments",
                    status: "pending"
                ),
                PlanningPhaseData(
                    id: "enriching",
                    label: "Enriching with microclimate weather forecasts and wind patterns for each proposed route corridor",
                    status: "pending"
                ),
                PlanningPhaseData(
                    id: "finalizing",
                    label: "Finalizing comprehensive route profiles with turn-by-turn directions and elevation profiles",
                    status: "pending"
                ),
            ],
            message: NavigatorMessage(
                id: "msg-001",
                sessionId: "session-001",
                body: "I'm currently analyzing your request for a coastal motorcycle adventure from Santa Cruz to Big Sur. I'm examining the topography of the region, identifying the most scenic and technically engaging roads, checking current road conditions and closures, analyzing weather patterns along different route corridors, and evaluating traffic patterns for various departure times. This comprehensive analysis ensures I can provide you with the absolute best routing options for your ride today.",
                timestamp: "2025-04-25T10:30:00Z",
                kind: "response",
                attachments: nil,
                detail: "I'm particularly focused on finding routes with optimal corner sequences and scenic overlooks while avoiding any construction or road maintenance areas.",
                pinned: false
            ),
            isThinking: true,
            showSlowApology: false,
            showCancelConfirm: false,
            showWarningChrome: false
        )
    }

    // MARK: - V01: Slow Planning

    private static func slowPlanningState() -> PlanningScreenState {
        PlanningScreenState(
            phases: [
                PlanningPhaseData(id: "parsing", label: "Parsing your request", status: "done"),
                PlanningPhaseData(id: "searching", label: "Searching routes", status: "active"),
                PlanningPhaseData(id: "drafting", label: "Drafting options", status: "pending"),
                PlanningPhaseData(id: "enriching", label: "Enriching with details", status: "pending"),
                PlanningPhaseData(id: "finalizing", label: "Finalizing your routes", status: "pending"),
            ],
            message: NavigatorMessage(
                id: "msg-001",
                sessionId: "session-001",
                body: "Looking for the best twisty roads between Santa Cruz and Big Sur...",
                timestamp: "2025-04-25T10:30:00Z",
                kind: "response",
                attachments: nil,
                detail: "Still scouting… this area is surprisingly complex.",
                pinned: false
            ),
            isThinking: true,
            showSlowApology: true, // V01: Show slow apology
            showCancelConfirm: false,
            showWarningChrome: false
        )
    }

    // MARK: - V02: Cancel Confirm

    private static func cancelConfirmState() -> PlanningScreenState {
        PlanningScreenState(
            phases: [
                PlanningPhaseData(id: "parsing", label: "Parsing your request", status: "done"),
                PlanningPhaseData(id: "searching", label: "Searching routes", status: "active"),
                PlanningPhaseData(id: "drafting", label: "Drafting options", status: "pending"),
                PlanningPhaseData(id: "enriching", label: "Enriching with details", status: "pending"),
                PlanningPhaseData(id: "finalizing", label: "Finalizing your routes", status: "pending"),
            ],
            message: NavigatorMessage(
                id: "msg-001",
                sessionId: "session-001",
                body: "Looking for the best twisty roads between Santa Cruz and Big Sur...",
                timestamp: "2025-04-25T10:30:00Z",
                kind: "response",
                attachments: nil,
                detail: nil,
                pinned: false
            ),
            isThinking: true,
            showSlowApology: false,
            showCancelConfirm: true, // V02: Show cancel confirm
            showWarningChrome: false
        )
    }

    // MARK: - V03: Single Candidate

    private static func singleCandidateState() -> PlanningScreenState {
        PlanningScreenState(
            phases: [
                PlanningPhaseData(id: "parsing", label: "Parsing your request", status: "done"),
                PlanningPhaseData(id: "searching", label: "Searching routes", status: "done"),
                PlanningPhaseData(id: "drafting", label: "Drafting options", status: "active"),
                PlanningPhaseData(id: "enriching", label: "Enriching with details", status: "pending"),
                PlanningPhaseData(id: "finalizing", label: "Finalizing your routes", status: "pending"),
            ],
            message: NavigatorMessage(
                id: "msg-001",
                sessionId: "session-001",
                body: "Found one solid option that matches your constraints.",
                timestamp: "2025-04-25T10:30:00Z",
                kind: "response",
                attachments: nil,
                detail: "Only one candidate meets your criteria — the coastal route via Hwy 1.",
                pinned: false
            ),
            isThinking: true,
            showSlowApology: false,
            showCancelConfirm: false,
            showWarningChrome: true // V03: Show warning chrome
        )
    }
}
